### PR TITLE
Task7

### PR DESCRIPTION
**Pull request by meis0n**

### CODE REVIEW

File: src/app/courses/course-data-page/course-data-page.component.ts [hunk 1]
```diff
@@ -10,4 +14,47 @@ export class CourseDataPageComponent {
...
+    private courseServise: CourseService,
+    private router: Router
+  ) {}
+
+  async ngOnInit(): Promise<void> {
+    this.mode = this.route.snapshot.data['mode'];
+
+    if (this.isEditMode) {
+      const id = this.route.snapshot.paramMap.get('id');
+      const data = await this.courseServise.getCourseById(id);
```
[the remark this answers is not in the document]
Comment (meis0n) [reply src/app/courses/course-data-page/course-data-page.component.ts]: i'll fix it during task9 complition

File: src/app/shared/components/breadcrumbs/breadcrumbs.component.ts [hunk 2]
```diff
@@ -1,8 +1,48 @@
...
+  }
+
+  buildBreadCrumb(route: ActivatedRoute = this.activatedRoute.root, url: string = '',
+                breadcrumbs: Array<Breadcrumb> = []): Array<Breadcrumb> {
+    const label = route.routeConfig && route.routeConfig.data && route.routeConfig.data[ 'breadcrumb' ];
+
+    const nextUrl = `${url}${route.snapshot.url.join('/')}/`;
+
+    const newBreadcrumbs = [ ...breadcrumbs, {
+      title: isFunction(label) ? label(route) : label,
```
[the remark this answers is not in the document]
Comment (meis0n) [reply src/app/shared/components/breadcrumbs/breadcrumbs.component.ts]: i want to create breadcrums, e.g "Video course #1234" by passsing the route obj

courses-routing.module.ts
image
image

File: src/app/courses/course-data-page/course-data-page.component.ts [hunk 3]
```diff
@@ -10,4 +14,47 @@ export class CourseDataPageComponent {
...
+
+  onCancel (): void {
+    this.router.navigateByUrl('courses');
+  }
+
+  async onSave (): Promise<void> {
+    if (this.isEditMode) {
+
+    }
+    else {
```
[the remark this answers is not in the document]
Comment (meis0n) [reply src/app/courses/course-data-page/course-data-page.component.ts]: done

File: src/app/courses/course-data-page/course-data-page.component.ts [hunk 4]
```diff
@@ -10,4 +14,47 @@ export class CourseDataPageComponent {
...
+      await this.courseServise.createCourse(this.data);
+    }
```
[the remark this answers is not in the document]
Comment (meis0n) [reply src/app/courses/course-data-page/course-data-page.component.ts]: done

File: src/app/app.module.ts [hunk 5]
```diff
@@ -6,25 +6,70 @@ import { AppRoutingModule } from './app-routing.module';
...
+          component: CourseDataPageComponent,
+          data: {
+            mode: 'edit',
+            breadcrumb: (route: ActivatedRoute) => `Video course ${route.snapshot.params.id}`
+          }
+        },
+      ]
+    },
+    {
+      path: '',
```
[the remark this answers is not in the document]
Comment (meis0n) [reply src/app/app.module.ts]: done

File: src/app/app.module.ts [hunk 6]
```diff
@@ -6,25 +6,70 @@ import { AppRoutingModule } from './app-routing.module';
...
+          path: '',
+          component: CoursesPageComponent,
+        },
+        {
+          path: 'new',
+          component: CourseDataPageComponent,
+          data: {
+            breadcrumb: 'New',
+            //Вопрос: в идеале я зотел бы видеть это чем-то вроде константы
+            // COURSE_DETAIL_MODES.EDIT|NEW, но как правильно их заимпортить в коммон модуль из своего модуля, отвечающего за страницы курсов?
```
[the remark this answers is not in the document]
Comment (meis0n) [reply src/app/app.module.ts]: yes, it looks as what i need

File: src/app/app.module.ts [hunk 7]
```diff
@@ -6,25 +6,70 @@ import { AppRoutingModule } from './app-routing.module';
...
+    path: '',
+    data: {
+      breadcrumb: 'Home',
+    },
+    children:[{
+      path: 'login',
+      component: LoginPageComponent
+    },
+    {
+      path: 'courses',
```
[the remark this answers is not in the document]
Comment (meis0n) [reply src/app/app.module.ts]: done

File: UI/src/app/courses/course-data-page/course-data-page.component.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  onCancel (): void {
+    this.router.navigateByUrl('courses');
+  }
+
+  async onSave (): Promise<void> {
+    // TODO: Изменить при выполнении task9
+    if (this.isEditMode) {
+      return;
+    }
+    else {
```
Comment (AinuraAliieva) [inline UI/src/app/courses/course-data-page/course-data-page.component.ts]: looks useless. we don't need else here

Comment (meis0n) [reply UI/src/app/courses/course-data-page/course-data-page.component.ts]: fixed in task9

**Pull request merged**